spyder/plugins/ipythonconsole/widgets/shell.py

```diff
@@ -318,8 +357,35 @@ def check_spyder_kernel_callback(self, reply):
         # Process kernel reply
```

Comment:
The docstring of this method is incorrect. I can directly leave a suggestion for it because it's not part of the diff, but I recommend to change it for:

```python
def check_spyder_kernel_callback(self, reply):
    """
    Check if the Spyder-kernels version is the right one after receiving it
    from the kernel.
    """
```